sunkit_image/nafe.py

```diff
@@ -0,0 +1,411 @@
+    parser.add_argument("-nafe",  "--activate_nafe",    help="activate NAFE", action="store_true")
+    parser.add_argument("-w",     "--nafe_weight",      help="NAFE weight",                             type=float)
+    parser.add_argument("-roi_p", "--ROI_start_point",  help="(row, col) of ROI top left corner pixel", type=int,  nargs='+')
+    parser.add_argument("-roi_f", "--ROI_frame",        help="(width, height) of ROI frame",            type=int,  nargs='+')
+    parser.add_argument("-p",     "--nproc",            help="number of processes for multiprocessing", type=int)
+    parser.add_argument("-hb",    "--hist_bins",        help="number of histogram bins",                type=int)
+
+    # Parameters (from ini file)
+    parser.add_argument("-nrs", "--NRSigma", help="noise reduction sigma", type=float, default=15.0)
+    # Contrast control
+    parser.add_argument("-gb", "--global_black", help="global weight for calculating the output black",    type=float, default=0.3)
+    parser.add_argument("-lb", "--local_black",  help="local roi weight for calculating the output black", type=float, default=0.7)
+    parser.add_argument("-gw", "--global_white", help="global weight for calculating the output white",    type=float, default=0.3)
+    parser.add_argument("-lw", "--local_white",  help="local roi weight for calculating the output white", type=float, default=0.7)
+    # Fuzzy neighborhood
+    parser.add_argument("-n", "--neighborhood_width", help="fuzzy neighborhood width",     type=int,   default=129)
+    parser.add_argument("-c", "--constants",          help="fuzzy neighborhood constants", type=float, default=[1 for _ in range(12)], nargs='+')
+
+    args = vars(parser.parse_args())
+    
+    imargin = tuple(args["input_margin"]) if args["input_margin"] is not None else None
+    omargin = tuple(args["output_margin"]) if args["output_margin"] is not None else None
+    roi_p = tuple(args["ROI_start_point"]) if args["ROI_start_point"] is not None else None
+    roi_f = tuple(args["ROI_frame"]) if args["ROI_frame"] is not None else None
+
+    input_map = sunpy.map.Map(args["image"])
+    nafe_image = nafe(input_map=input_map, imargin=imargin,
+                      gamma=args["gamma"], omargin=omargin,
+                      nafe=args["activate_nafe"],
+                      nafe_weight=args["nafe_weight"], roi_p=roi_p,
+                      roi_f=roi_f, nproc=args["nproc"],
+                      hbins=args["hist_bins"], nrs=args["NRSigma"],
+                      gb=args["global_black"], lb=args["local_black"],
+                      gw=args["global_white"], lw=args["local_white"],
+                      n=args["neighborhood_width"], c=args["constants"])
+
+    figure, axes = plt.subplots(1, 2)
+    axes[0].imshow(input_map.data[1000:1400, 1000:1400], cmap='gray')
+    axes[0].set_title('Original Image')
+    axes[1].imshow(nafe_image, cmap='gray')
+    axes[1].set_title('Enhanced Image')
+    plt.show()
```

Comment:
This can all go. We don't want to be able to call the function like this. All those `arser.add_argument(` should the parameters in the corresponding doc strings. 

This could be turned into a seperate gallery example.

Reply:
It's basically for debugging purposes but will apply for final PR.

Reply:
Though I agree with @nabobalis and probably this wouldn't be the place. I imagine sunkit-image could have some command line programs that run some of these functions. This would need, however some extra additions to the `setup.py`.

Reply:
Will remove from file for now and investigate where to migrate after finishing testing.